shell/platform/android/io/flutter/embedding/android/FlutterImageView.java

```diff
@@ -41,13 +41,33 @@
   @Nullable private Bitmap currentBitmap;
   @Nullable private FlutterRenderer flutterRenderer;
 
+  public enum SurfaceKind {
+    /** Displays the background canvas. */
+    background,
+
+    /** Displays the overlay surface canvas. */
+    overlay,
+  }
```

Comment:
The java/c++ flows are very different depending on the surface kind. Maybe, the code can be refactored to have a unified flow.  This is out of scope at the moment though.